### ISSUE TITLE
Tasks with more than one item in `Machines` parameter cause various failures

### ISSUE THREAD

**Issue by mariocatch** [Microsoft/vsts-tasks#2270]
After 8/12/2016, and grabbing the new Agent for on-prem VSO agents, we're getting constant failures on a *Windows Machine File Copy* task

```
2016-08-12T20:23:03.0199379Z ##[error]System.Management.Automation.RemoteException: The term 'Get-LocalizedString' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
 
2016-08-12T20:23:03.0199379Z ##[error]System.Management.Automation.RemoteException: Exception calling "LoadFrom" with "1" argument(s): "Could not load file or assembly 'file:///C:\Agent\externals\vstshost\Microsoft.TeamFoundation.DistributedTask.Task.LegacySDK.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)"
 
2016-08-12T20:23:03.0199379Z ##[error]System.Management.Automation.RemoteException: Could not load file or assembly 'file:///C:\Agent\externals\vstshost\Microsoft.TeamFoundation.DistributedTask.Task.LegacySDK.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)
```

**Edit 1** 

This seems to occur when passing in at least 2 Machines into the task parameters.

**Edit 2**

The above issue is resolved by separating out the Machines into 2 Windows Machine File Copy tasks on the Release Definition.

However, there is a new issue now:

```
[error]System.Management.Automation.ActionPreferenceStopException: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: Exception calling "LoadFrom" with "1" argument(s): "Could not load file or assembly 'file:///C:\Agent\externals\vstshost\Microsoft.TeamFoundation.DistributedTask.Task.LegacySDK.dll' or one of its dependencies. Operation is not supported. (Exception from HRESULT: 0x80131515)"
```

Note, these issues are only happening after the updates that seem to have been made on 8/12/2016. We are using the latest Agent executable, for on-prem.

**Edit 3**

The above error is resolved by splitting out the remaining Tasks into individual Tasks wherever we were using more than one item in a Machines input before.

Tested with Windows Machine File Copy task and with Remote Powershell task.

To summarize, the issue simply seems to be related to Tasks that accept a Machines argument. If we have 2 items in that list, it breaks. If we separate out that Task into 2 tasks (each with their own machine) then it works.

**Comment by chshrikh** [comment]
Hi @mariocatch , could you please let us know which OS you are using for running you agent ?

**Comment by chshrikh** [comment]
@mariocatch thanks for reporting it. We are fixing it with next release. To unblock yourself please follow below steps

1)	Go to agent box
2)	Launch the cmd with admin privileges
3)	Stop your running agent
4)	Go to your agent directory ( where you can find the config.cmd  e.g. c:\agent)
5)	Run this command –

powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "$VerbosePreference = %VERBOSE_ARG% ; Get-ChildItem -LiteralPath '%~dp0' | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null ; Get-ChildItem -Recurse -LiteralPath '%~dp0bin', '%~dp0externals' | Where-Object { $_ -match '\.(ps1|psd1|psm1)$' } | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null ; Get-ChildItem -LiteralPath '%~dp0externals\vstsom', '%~dp0externals\vstshost' | Where-Object { $_ -match '\.(dll|exe)$' } | ForEach-Object { Write-Verbose ('Unblock: {0}' -f $_.FullName) ; $_ } | Unblock-File | Out-Null"

**Comment by kmkumaran** [comment]
@mariocatch based on response from @chshrikh I am marking this issue as resolved. Feel free to reopen if you are still blocked.
